src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -313,94 +313,133 @@ protected static void DisplayManifestPreview(Manifests manifests)
+        /// <returns>A boolean value indicating whether the OAuth login flow was successful.</returns>
+        protected async Task<bool> GetTokenFromOAuth()
+        {
+            Logger.DebugLocalized(nameof(Resources.GitHubAccountMustBeLinked_Message));
+            Logger.DebugLocalized(nameof(Resources.ExecutionPaused_Message));
+            Console.WriteLine();
+            this.GitHubToken = await GitHubOAuthLoginFlow();
+
+            if (string.IsNullOrEmpty(this.GitHubToken))
+            {
+                // User must've cancelled OAuth flow, we can't proceed successfully
+                Logger.WarnLocalized(nameof(Resources.NoTokenResponse_Message));
+                return false;
+            }
+
+            this.StoreTokenInCache();
+            Logger.DebugLocalized(nameof(Resources.ResumingCommandExecution_Message));
+            return true;
+        }
+
+        /// <summary>
+        /// If the provided token is valid, stores the token in cache.
+        /// </summary>
+        /// <returns>Returns a boolean value indicating whether storing the token in cache was successful.</returns>
+        protected bool StoreTokenInCache()
+        {
+            try
+            {
+                Logger.Trace("Writing token to cache");
+                GitHubOAuth.WriteTokenCache(this.GitHubToken);
+                Logger.InfoLocalized(nameof(Resources.StoringToken_Message));
+            }
+            catch (Exception ex)
+            {
+                // Failing to cache the token shouldn't be fatal.
+                Logger.WarnLocalized(nameof(Resources.WritingCacheTokenFailed_Message), ex.Message);
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Verifies if the GitHub token has valid access.
+        /// </summary>
+        /// <returns>A boolean value indicating whether the GitHub token had valid access.</returns>
+        protected async Task<bool> CheckGitHubToken()
+        {
+            var client = new GitHub(this.GitHubToken, this.WingetRepoOwner, this.WingetRepo);
+
+            try
+            {
+                Logger.Trace("Checking repo access using provided token");
+                await client.CheckAccess();
+                Logger.Trace("Access check was successful, proceeding");
+            }
+            catch (Exception e)
+            {
+                if (e is AuthorizationException)
                 {
-                    Logger.ErrorLocalized(nameof(Resources.Error_Prefix), e.Message);
-                    Logger.ErrorLocalized(nameof(Resources.InvalidTokenError_Message));
-                    return false;
+                    Logger.WarnLocalized(nameof(Resources.InvalidGitHubToken_Message));
                 }
-                else
+                else if (e is RateLimitExceededException)
                 {
-                    throw;
+                    Logger.WarnLocalized(nameof(Resources.RateLimitExceeded_Message));
```

Comment:
Do we want to automatically launch the oauth flow if we're rate limited?

Reply:
Talked offline. I think we shouldn't as a clear error message with instructions on how to bypass this error should be sufficient and user friendly